syft/cataloger/common/generic_cataloger.go

```diff
@@ -34,73 +34,82 @@ func NewGenericCataloger(pathParsers map[string]ParserFn, globParsers map[string
 }
 
 // Name returns a string that uniquely describes the upstream cataloger that this Generic Cataloger represents.
-func (a *GenericCataloger) Name() string {
-	return a.upstreamCataloger
+func (c *GenericCataloger) Name() string {
+	return c.upstreamCataloger
 }
 
 // register pairs a set of file references with a parser function for future cataloging (when the file contents are resolved)
-func (a *GenericCataloger) register(files []file.Reference, parser ParserFn) {
-	a.selectedFiles = append(a.selectedFiles, files...)
+func (c *GenericCataloger) register(files []file.Reference, parser ParserFn) {
+	c.selectedFiles = append(c.selectedFiles, files...)
 	for _, f := range files {
-		a.parsers[f] = parser
+		c.parsers[f] = parser
 	}
 }
 
 // clear deletes all registered file-reference-to-parser-function pairings from former SelectFiles() and register() calls
-func (a *GenericCataloger) clear() {
-	a.selectedFiles = make([]file.Reference, 0)
-	a.parsers = make(map[file.Reference]ParserFn)
+func (c *GenericCataloger) clear() {
+	c.selectedFiles = make([]file.Reference, 0)
+	c.parsers = make(map[file.Reference]ParserFn)
+}
+
+func (c *GenericCataloger) Catalog(resolver scope.Resolver) ([]pkg.Package, error) {
+	fileSelection := c.selectFiles(resolver)
+	contents, err := resolver.MultipleFileContentsByRef(fileSelection...)
```

Comment:
This is probably the wrong spot for this comment — but I'd suggest making the parameter for `MultipleFileContentsByRef` not variadic, if it's true that its argument is always a slice that's needing to be expanded with `...`

Reply:
I don't think there is a performance impact here and allowing for variadic args enables not needing to construct a list when providing multiple arguments.

Reply:
Yeah, this was more of an ergonomics comment, there's certainly no performance impact. It's okay to keep this as is for now, but let's stay aware of ergonomic indicators as we move forward — e.g., if callers never end up needing to construct slices anyway (because they already have the data in slice form) then the choice to make this variadic is actually adding work to consumers